lib/models/fhir_questionnaire/questionnaire_model.dart

```diff
@@ -61,10 +63,11 @@ class QuestionnaireModel {
   }
 
   /// saves the Questionnaire locally
-  Future<Resource> saveResponses(List<UserResponse> responses) async {
+  Future saveResponses(List<UserResponse> responses) async {
```

Comment:
suggest keeping `Future<Resource>` so that the future knows what it is returning

Reply:
This saves QuestionnaireResponse, plus any associated conditions and observations, so it's not just a single Resource it would be returning. I can have it return a bundle of all of the resources saved, I can have it return a list of the resources, or return Either<Failure, Unit>, where failure would be the error message, and unit is nothing, it just tells you that function completed properly.